src/nvim/api/vim.c

```diff
@@ -1245,6 +1245,179 @@ ArrayOf(Dictionary) nvim_get_keymap(String mode)
   return keymap_array(mode, NULL);
 }
 
+/// Sets a |mapping| for the given mode.
+///
+/// Note that isn't possible to specify a target buffer when opts contains the
```

Comment:
We would like a separate `nvim_buf_set_keymap` for this. Compare how `nvim_get_keymap` and `nvim_buf_get_keymap` is implemented. Most of the docs can stay at `nvim_set_keymap`, and the `nvim_buf` variant can just reference it.

Reply:
While I would love to implement this, I don't think that I have the time to do so, at least for now. Would it be possible to split this off into a different issue/PR?

Reply:
New API functions should ideally not hardcode `curbuf` usage. I don't think this should be so difficult to achieve. Make `do_map` (or the second half, if the split suggestion is followed) take `buf_T *buf` arg and use it everywhere instead of `curbuf` inside of the function.